### PR TITLE
Sizing calculator integration

### PR DESCRIPTION
**Pull request by shaharuk-yb**
The Pull Request contains following implementation:
1. sizing and sharding business logic
2. integration with voyager and assessment framework
3. reporting in json and html format

example of the json output:
[assessmentReport.json](https://github.com/yugabyte/yb-voyager/files/15011783/assessmentReport.json)

html report (screenshot of sizing and sharding section only):
Screenshot 2024-04-17 at 7 10 38 PM

Notes:
- based on further discussions, the business logic may change over time

### CODE REVIEW

**Review by sanyamsinghal — COMMENTED** [review]
Need to connect for understanding the `sizing.go` implementation. Rest all files are fine with minor comments.

File: yb-voyager/src/migassessment/sizing.go [hunk 1]
```diff
@@ -16,29 +16,515 @@ limitations under the License.
...
-	// add any sizing specific parameters required from user here
-	// dummy params for now
-	UseNvme           bool `toml:"use_nvme"`
-	MultiAzDeployment bool `toml:"multi_az_deployment"`
+type SourceDBMetadata struct {
+	SchemaName string        `json:"schema_name"`
+	ObjectName string        `json:"object_name"`
+	RowCount   sql.NullInt64 `json:"row_count,string"`
+	//ColCount        int64  `json:"col_count,string"`
+	//ReadsPerSec     int64  `json:"reads_per_second,string"`
```
Comment (amit-yb) [inline yb-voyager/src/migassessment/sizing.go:36]: Delete commented code from everywhere.

File: yb-voyager/src/migassessment/sizing.go [hunk 2]
```diff
@@ -16,29 +16,515 @@ limitations under the License.
...
+for tables and indexes. It also returns the total size of source db in GB
+Returns:
+
+	[]SourceDBMetadata: all table objects from source db
+	[]SourceDBMetadata: all index objects from source db
+	float64: total size of source db
+*/
+func loadSourceMetadata() ([]SourceDBMetadata, []SourceDBMetadata, float64) {
+	err := ConnectSourceMetaDatabase(GetDBFilePath())
+	checkErr(err)
```
Comment (amit-yb) [inline yb-voyager/src/migassessment/sizing.go:71]: Get rid of this `checkErr()`. Instead return an `error` from the functions.

**Review by amit-yb — APPROVED** [review]
LGTM with minor nits.

File: yb-voyager/src/migassessment/common.go [hunk 3]
```diff
@@ -130,6 +136,45 @@ func LoadAssessmentParams(userInputFpath string) error {
...
+		log.Fatal(err)
+	}
+}
+
+func checkInternetAccess() (ok bool) {
+	_, err := http.Get("http://clients3.google.com/generate_204")
+	return err == nil
+}
+
+func checkLocalFileExists(filePath string) bool {
```
[the remark this answers is not in the document]
Comment (shaharuk-yb) [reply yb-voyager/src/migassessment/common.go]: done

File: yb-voyager/src/migassessment/common.go [hunk 4]
```diff
@@ -16,34 +16,40 @@ limitations under the License.
...
+
+type Report struct {
+	ColocatedTables                 []string
+	ColocatedReasoning              string
+	ShardedTables                   []string
+	NumNodes                        float64
+	VCPUsPerInstance                float64
+	MemoryPerInstance               float64
+	OptimalSelectConnectionsPerNode int64
+	OptimalInsertConnectionsPerNode int64
```
[the remark this answers is not in the document]
Comment (shaharuk-yb) [reply yb-voyager/src/migassessment/common.go:43]: No, this is incorrect. We have evolved from the initial discussions on how the reporting would look like. I think we had couple of discussions around the same. This is how the report will look like for now.

**Pull request merged**